ketcher-autotests/tests/Structure-Creating-&-Editing/SMARTS-attributes/Bond-attributes/bond-attributes.spec.ts

```diff
@@ -0,0 +1,209 @@
+
+  test('Setting bond type - any', async ({ page }) => {
+    await setBondType(page, 'Any-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(page, defaultFileFormat, '[#6](-[#6])(-[#6])~[#6]');
+  });
+
+  test('Setting bond type - hydrogen', async ({ page }) => {
+    await setBondType(page, 'Hydrogen-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(page, defaultFileFormat, '[#6](-[#6])(-[#6])[#6]');
+    await checkSmartsWarnings(page);
+  });
+
+  test('Setting bond type - single/double', async ({ page }) => {
+    await setBondType(page, 'Single/Double-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6])!:;-,=[#6]',
+    );
+  });
+
+  test('Setting bond type - single/aromatic', async ({ page }) => {
+    await setBondType(page, 'Single/Aromatic-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6])-,:[#6]',
+    );
+  });
+
+  test('Setting bond type - double/aromatic', async ({ page }) => {
+    await setBondType(page, 'Double/Aromatic-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6])=,:[#6]',
+    );
+  });
+
+  test('Setting bond type - dative', async ({ page }) => {
+    await setBondType(page, 'Dative-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(page, defaultFileFormat, '[#6](-[#6])(-[#6])[#6]');
+    await checkSmartsWarnings(page);
+  });
+
+  // Tests for bond topology:
+
+  test('Setting bond topology - ring', async ({ page }) => {
+    await setBondTopology(page, 'Ring-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6])-;@[#6]',
+    );
+  });
+
+  test('Setting bond topology - chain', async ({ page }) => {
+    await setBondTopology(page, 'Chain-option');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6])-;!@[#6]',
+    );
+  });
+
+  // Custom query for bond
+
+  test('Setting custom query - any OR double', async ({ page }) => {
+    test.fail();
+    /**
+     * This test will fail until https://github.com/epam/Indigo/issues/1372 is fixed
+     */
+    await setCustomQueryForBond(page, '~,=');
+    await pressButton(page, 'Apply');
+    await takeEditorScreenshot(page);
+    await checkSmartsValue(
+      page,
+      defaultFileFormat,
+      '[#6](-[#6])(-[#6])~,=[#6]',
+    );
+  });
+});
```

Comment:
This test should be here or we can move it to 'custom-query.spec.ts'y?

Reply:
'custom-query.spec.ts' has tests for custom query only for atoms (different code at beforeEach for example) while this is custom query for bond. I can create another file for that case but I thought it would be better to keep it here instead of creating file with one test case.